src/topology/algebra/module.lean

```diff
@@ -623,4 +713,78 @@ by { ext x, refl }
 @[simp] theorem symm_symm_apply (e : M ≃L[R] M₂) (x : M) : e.symm.symm x = e x :=
 rfl
 
+/-- Create a `continuous_linear_equiv` from two `continuous_linear_map`s that are
+inverse of each other. -/
+def of_inverse (f₁ : M →L[R] M₂) (f₂ : M₂ →L[R] M) (h₁ : function.left_inverse f₂ f₁)
+  (h₂ : function.right_inverse f₂ f₁) :
+  M ≃L[R] M₂ :=
+{ to_fun := f₁,
+  continuous_to_fun := f₁.continuous,
+  inv_fun := f₂,
+  continuous_inv_fun := f₂.continuous,
+  left_inv := h₁,
+  right_inv := h₂,
+  .. f₁ }
+
+@[simp] lemma of_inverse_apply (f₁ : M →L[R] M₂) (f₂ h₁ h₂ x) :
+  of_inverse f₁ f₂ h₁ h₂ x = f₁ x :=
+rfl
+
+@[simp] lemma symm_of_inverse (f₁ : M →L[R] M₂) (f₂ h₁ h₂) :
+  (of_inverse f₁ f₂ h₁ h₂).symm = of_inverse f₂ f₁ h₂ h₁ :=
+rfl
+
+section
+variables (R) [topological_space R] [topological_module R R]
+
+/-- Continuous linear equivalences `R ≃L[R] R` are enumerated by `units R`. -/
+def equiv_units : units R ≃ (R ≃L[R] R) :=
+{ to_fun := λ u, of_inverse
+    (continuous_linear_map.smul_right 1 ↑u)
+    (continuous_linear_map.smul_right 1 ↑u⁻¹)
+    (λ x, by simp) (λ x, by simp),
+  inv_fun := λ e, ⟨e 1, e.symm 1,
+    by rw [← smul_eq_mul, ← map_smul, smul_eq_mul, mul_one, symm_apply_apply],
+    by rw [← smul_eq_mul, ← map_smul, smul_eq_mul, mul_one, apply_symm_apply]⟩,
+  left_inv := λ u, units.ext $ by simp,
+  right_inv := λ e, ext₁ $ by simp }
+
+variable {R}
+
+@[simp] lemma equiv_units_apply (u : units R) (x : R) : equiv_units R u x = x * u := rfl
+
+@[simp] lemma equiv_units_apply_symm (u : units R) (x : R) :
+  (equiv_units R u).symm x = x * ↑u⁻¹ := rfl
+
+@[simp] lemma equiv_units_symm_apply (e : R ≃L[R] R) :
+  ↑((equiv_units R).symm e) = e 1 :=
+rfl
+
+end
+
+variables [topological_add_group M]
+
+open continuous_linear_map (id fst snd subtype_val mem_ker)
+
+/-- A pair of continuous linear maps such that `f₁ ∘ f₂ = id` generates a continuous
+linear equivalence `e` between `M` and `M₂ × f₁.ker` such that `(e x).2 = x` for `x ∈ f₁.ker`,
+`(e x).1 = f₁ x`, and `(e (f₂ y)).2 = 0`. The map is given by `e x = (f₁ x, x - f₂ (f₁ x))`. -/
+def of_right_inverse (f₁ : M →L[R] M₂) (f₂ : M₂ →L[R] M) (h : function.right_inverse f₂ f₁) :
```

Comment:
Maybe `equiv_of_right_inverse` here? Perhaps I'm missing an open namespace, but I think this name doesn't have enough identifying information.

Reply:
It's inside `continuous_linear_equiv` but I can add `equiv_` if you want (AFAIR we do it in `equiv.*`)